device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubSendTask.java

```diff
@@ -28,12 +43,20 @@ public void run()
         }
         // Codes_SRS_IOTHUBSENDTASK_11_005: [The function shall not crash because of an IOException thrown by the transport.]
         // Codes_SRS_IOTHUBSENDTASK_11_008: [The function shall not crash because of any error or exception thrown by the transport.]
+        // JMayrbaeurl 2017-03-10: Bad idea to catch any Throwable here!!!
         catch (Throwable e)
         {
+        	// JMayrbaeurl 2017-03-10: Should be replaced with real logging setup
             System.out.println(e.toString() + ": " + e.getMessage());
-            for (StackTraceElement el : e.getStackTrace())
+            
+            // Only output full stack trace on verbose Error messaging configuration
+            if (this.verboseErrorMessaging) 
             {
-                System.out.println(el);
+	            for (StackTraceElement el : e.getStackTrace())
+	            {
+	            	// JMayrbaeurl 2017-03-10: Should be replaced with real logging setup
+	                System.out.println(el);
```

Comment:
I think we have logger use in this module and would recommend using logger here instead of println. So that interested users can debug it using log.
And if we go the route of logger you woudn't really need verboseErrMessaging anymore. By default you can always write fatal error message in this case.